### PR TITLE
feat: Introduce Formbricks CX

### PR DESCRIPTION
**Pull request by gupta-piyush19**
## What does this PR do?
Introduce Formbricks CX

Fixes https://github.com/formbricks/internal/issues/356

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

## Release Notes

- **New Features**
  - Added multiple survey templates for enhanced survey creation, including NPS, CSAT, and more.
  - Introduced a new interface for selecting between "Web app" and "Mobile app" embedding options.
  - Enhanced survey embedding options with static iframe and dynamic pop-up choices.

- **Improvements**
  - Improved user experience with clear guidance and instructions tailored to different application types.
  - Enhanced flexibility in onboarding options and question type selections, including conditional rendering based on "CX mode."
  - Added utility functions for customizing survey templates with product names.

- **Bug Fixes**
  - Corrected URL path for notifications settings to ensure proper routing.

### CODE REVIEW

**Comment by gupta-piyush19** [comment]
@jobenjada Please review the new onboarding and the changes in the survey editor.

**Review by jobenjada — CHANGES_REQUESTED** [review]
hey Piyush!

Looks great :) Really like the stripped down survey editor, makes it a lot easier to get to the first finalized survey 💪 

A few things:

Here, please use the toggle we're also using in the Survey Editor e.g. to determine which validation a text input has:
image

image

----------

As discussed on Discord: Is it possible here to set the toggle to default on and hide the toggle and the two text links below the background color?
image

------------

And pls review my code changes and resolve the merge conflict :)

File: apps/web/app/(app)/(onboarding)/environments/[environmentId]/xm-templates/lib/utils.ts
Comment (jobenjada) [inline apps/web/app/(app)/(onboarding)/environments/[environmentId]/xm-templates/lib/utils.ts:1]: do we need this duplicate?

File: apps/web/app/(app)/(onboarding)/environments/[environmentId]/xm-templates/page.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const product = await getProductByEnvironmentId(environment.id);
+  if (!product) {
+    throw new Error("Product not found");
+  }
+
+  return (
+    <div className="flex min-h-full min-w-full flex-col items-center justify-center space-y-12">
+      <Header title="What kind of feedback would you like to get?" />
+      <XMTemplateList product={product} user={user} environmentId={environment.id} />
+      <Button
```
Comment (jobenjada) [inline apps/web/app/(app)/(onboarding)/environments/[environmentId]/xm-templates/page.tsx:43]: it's missing the condition checking for product.length !== 0

Comment (jobenjada) [reply apps/web/app/(app)/(onboarding)/environments/[environmentId]/xm-templates/page.tsx:43]: {products.length >= 1 && (

**Review by jobenjada — CHANGES_REQUESTED** [review]
last tweaks, some merge conflicts, then LGTM 🚀

File: apps/web/app/(app)/(onboarding)/environments/[environmentId]/xm-templates/lib/utils.ts
Comment (gupta-piyush19) [reply apps/web/app/(app)/(onboarding)/environments/[environmentId]/xm-templates/lib/utils.ts:1]: Yeah, atleast the `replacePresetPlaceholders` handler, as the XM templates have a different schema than the normal templates

File: apps/web/app/(app)/(onboarding)/environments/[environmentId]/xm-templates/page.tsx (hunk 1 above)
Comment (gupta-piyush19) [reply apps/web/app/(app)/(onboarding)/environments/[environmentId]/xm-templates/page.tsx:43]: Do we need this check here? We are only fetching a single product and throwing an error if it doesn't exist.

**Review by jobenjada — APPROVED** [review]
whoopwhooop

**Pull request merged**